ha/core/support_bundle/ha_bundle.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import os
+
+from cortx.utils.schema.payload import Tar
+from cortx.utils.process import SimpleProcess
+from cortx.utils.log import Log
+
+from ha import const
+from ha.core.error import SupportBundleError
+
+class HABundle:
+
+    def __init__(self):
+        """
+        This Class generates the Support Bundle for Component HA.
+        This method will generate bundle for CSM and include the logs in it.
+        :param command: Csm_cli Command Object :type: command
+        :return:
+        """
+        # TODO Add setup detection logic
+        self._setup_type = "ldr"
+        self._id = None
+        self._path = None
+
+    def process_request(self, action, args, output):
+        """
+        Process support Bundle request
+
+        Args:
+            action (string): Action to be taken for support bundle.
+            args (dict): Parameter for support bundle
+        """
+        self._output = output
+        self._id = args.id
+        self._path = args.path
+        getattr(self, args.bundle_action)()
+
+    def create(self):
+        """
+        Create support bundle on given path.
+        """
+        if self._setup_type == "ldr":
+            self._add_more_log()
+        export_path = self._export_logs(self._check_files())
```

Comment:
How do we make sure that the HA logs bundle gets into the overall support bundle?

Reply:
First all log get captured

export is last cmd
and this log will get captured in cortxha.log

def _check_files(self):

this function is created to verify and log support bundle

Reply:
ok